lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -548,20 +549,38 @@ class ReportUIFeatures {
   /**
    * Opens a new tab to the treemap app and sends the JSON results using postMessage.
    * @param {LH.Result} json
+   * @param {'postMessage'|'url'} method
+   * @protected
    */
-  static openTreemap(json) {
+  static openTreemap(json, method = 'postMessage') {
     const treemapData = json.audits['script-treemap-data'].details;
     if (!treemapData) {
       throw new Error('no script treemap data found');
     }
 
-    const windowName = `treemap-${json.requestedUrl}`;
-    /** @type {LH.Treemap.Options} */
-    const treemapOptions = {
-      lhr: json,
+    // Only send the minimum lhr needed for the treemap app.
+    /** @type {RecursivePartial<LH.Treemap.Options>} */
+    const treemapOptionsTrimmed = {
```

Comment:
can this object's shape be the type of `LH.Treemap.Options`? The type within `main.js` will be silently broken, otherwise